apps/settings/js/panels/firefox_accounts/firefox_accounts.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+define(function(require) {
+  var normalizer = require('shared/text_normalizer');
+  var fxaHelper = require('shared/fxa_iac_client');
+  var dialogService = require('modules/dialog_service');
+
+  var FirefoxAccounts = function() {
+    this._boundRefreshStatus =
+      this._boundRefreshStatus || this.refreshStatus.bind(this);
+  };
+
+  FirefoxAccounts.prototype = {
+    onInit: function fxa_init(elements) {
+      this._elements = elements;
+    },
+
+    onBeforeShow: function fxa_beforeShow() {
+      fxaHelper.addEventListener('onlogin', this._boundRefreshStatus);
+      fxaHelper.addEventListener('onverified', this._boundRefreshStatus);
+      fxaHelper.addEventListener('onlogout', this._boundRefreshStatus);
+      this._boundRefreshStatus();
+    },
+
+    onBeforeHide: function fxa_beforeHide() {
+      fxaHelper.removeEventListener('onlogin', this._boundRefreshStatus);
+      fxaHelper.removeEventListener('onverified', this._boundRefreshStatus);
+      fxaHelper.removeEventListener('onlogout', this._boundRefreshStatus);
+    },
+
+    refreshStatus: function fxa_refreshStatus() {
+      fxaHelper.getAccount(
+        this._onFxAccountStateChange.bind(this),
+        this._onFxAccountError.bind(this));
+    },
+
+    // if e == null, user is logged out.
+    // if e.verified, user is logged in & verified.
+    // if !e.verified, user is logged in & unverified.
+    _onFxAccountStateChange: function fxa_onFxAccountStateChange(e) {
+      var email = e ? normalizer.escapeHTML(e.email) : '';
+      if (!e) {
+        this._hideLoggedInPanel();
+        this._hideUnverifiedPanel();
+        this._showLoggedOutPanel();
```

Comment:
Could we simplify those 6 methods to something like
```
showLoggedInPanel(false, email)
showUnverifiedPanel(false, email)
showLoggedOutPanel(true)
```

is it simpler or make it more complex?